### PR TITLE
fix: proxy detection needs to use session-relative plugins

### PR DESCRIPTION
**Pull request by jlipps**
fix a bug that snuck through in my last refactor

### CODE REVIEW

File: lib/appium.js [hunk 1]
```diff
@@ -360,16 +360,12 @@ class AppiumDriver extends BaseDriver {
...
+   * Get the appropriate plugins for a session (or sessionless plugins)
    *
-   * @param {string} cmd - the name of the command to find a plugin to handle
-   * @param {string} sessionId - the particular session for which to find a plugin, or null if
-   * sessionless
+   * @param {string} sessionId - the sessionId (or mull) to use to find plugins
+   * @returns {Array} - array of plugin instances
    */
-  pluginsToHandleCmd (cmd, sessionId) {
+  pluginsForSession (sessionId) {
```
Comment (mykola-mokhnach) [inline lib/appium.js:368]: sessionId = null

File: lib/appium.js [hunk 2]
```diff
@@ -360,16 +360,12 @@ class AppiumDriver extends BaseDriver {
...
-   * To get plugins for a command, we either get the plugin instances associated with the
-   * particular command's session, or in the case of sessionless plugins, pull from the set of
-   * plugin instances reserved for sessionless commands (and we lazily create plugin instances on
-   * first use)
+   * Get the appropriate plugins for a session (or sessionless plugins)
    *
-   * @param {string} cmd - the name of the command to find a plugin to handle
-   * @param {string} sessionId - the particular session for which to find a plugin, or null if
-   * sessionless
+   * @param {string} sessionId - the sessionId (or mull) to use to find plugins
```
Comment (mykola-mokhnach) [inline lib/appium.js:365]: `{?string}`

mull -> null

File: lib/appium.js [hunk 3]
```diff
@@ -382,9 +378,24 @@ class AppiumDriver extends BaseDriver {
...
+  }
+
+  /**
+   * To get plugins for a command, we either get the plugin instances associated with the
+   * particular command's session, or in the case of sessionless plugins, pull from the set of
+   * plugin instances reserved for sessionless commands (and we lazily create plugin instances on
+   * first use)
+   *
+   * @param {string} cmd - the name of the command to find a plugin to handle
+   * @param {string} sessionId - the particular session for which to find a plugin, or null if
```
Comment (mykola-mokhnach) [inline lib/appium.js:391]: `{?string}`

**Review by KazuCocoa — APPROVED** [review]
lg overall (only docstring?)

**Comment by jlipps** [comment]
published in beta 12

**Pull request merged**